### DISCUSSION TITLE
Installing openssl@1.0.0 - cannot load such file -- brewkit

### DISCUSSION THREAD

**Discussion by stephlaf** [Homebrew/discussions#732] [Everyday usage] [answered]
After installing `redis` using `build-from-source`, I am trying to install openssl@1.0.0. 

I extracted it to a local tap, but when trying to install, I get the error:
`cannot load such file -- brewkit`

Line 1 of the `.rb` file is `require 'brewkit'`

I can't find anything about this `brewkit` anywhere. Anybody can help on this?

Thanks

OS: High Sierra 10.13.6

**Comment by alebcay** [comment, marked as answer]
`brewkit` is what the original Homebrew (at least 5-6+ years ago now) used. The internal structure of Homebrew has changed greatly since then.

The formula you've extracted is extremely old - note that `brew extract` doesn't always create an installable formula; all it does is dig out the last version of the formula you requested from the Git history. In this case, Homebrew evidently hasn't shipped `openssl` 1.0.0 for a very long time now.

You may find the basic "modern" structure of a formula shown [here](https://docs.brew.sh/Formula-Cookbook#grab-the-url) to be a good starting point on how to edit the extracted formula into something that is installable today. The [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook) is also generally a helpful reference for authoring formulae.

**Reply by stephlaf** [reply]
Thank you very much for your explanations, @alebcay  🙏🏻
